src/validate-npm-package/validate-npm-package.ts

```diff
@@ -0,0 +1,95 @@
+// validate-npm-package/validate-npm-package.ts
+
+import os from 'node:os';
+import path from 'node:path';
+import { promises as fs } from 'node:fs';
+import childProcess from 'node:child_process';
+import util from 'node:util';
+
+import debug from 'debug';
+import { v4 as uuid } from 'uuid';
+
+import { addNPMRCFile } from '../publish-beta/publish';
+
+interface PackageJson {
+  type?: string;
+  service?: {
+    api: {
+      root: string;
+      endpoints: string[];
+    };
+  };
+}
+
+const exec = util.promisify(childProcess.exec);
+const log = debug('github-actions:validate-npm-package');
+
+async function obtainNpmPackage(packageVersion: string, workFolder: string): Promise<void> {
+  await addNPMRCFile(workFolder);
+
+  const fullCommandLine = `npm pack ${packageVersion}`;
+  log('obtainNpmPackage - fullCommandLine', fullCommandLine);
+
+  const execResult = await exec(fullCommandLine, { cwd: workFolder });
+  log('obtainNpmPackage - execResult', execResult);
+}
+
+async function unpackNpmPackage(packageTarballFilename: string, workFolder: string): Promise<string> {
+  const fullCommandLine = `tar zxvf ${packageTarballFilename}.tgz`;
+  log('unpackNpmPackage - fullCommandLine', fullCommandLine);
+
+  const execResult = await exec(fullCommandLine, { cwd: workFolder });
+  log('unpackNpmPackage - execResult', execResult);
+
+  return `${workFolder}/package`;
+}
+
+async function installNpmDependencies(packageFolder: string): Promise<void> {
+  await addNPMRCFile(packageFolder);
+
+  const fullCommandLine = `npm i --ignore-scripts`;
+  log('installNpmDependencies - fullCommandLine', fullCommandLine);
+
+  const execResult = await exec(fullCommandLine, { cwd: packageFolder });
+  log('installNpmDependencies - execResult', execResult);
+}
+
+async function verifyImportEntryPoints(packageFolder: string): Promise<void> {
+  const packageJson = JSON.parse(await fs.readFile(`${packageFolder}/package.json`, 'utf8')) as PackageJson;
+  const isEsm = packageJson.type === 'module';
+  const bundleFolder = isEsm ? 'dist-mjs' : 'dist';
```

Comment:
if `npm install` isn't an option, think it would be better to simulate what node does, i.e. read the package.json and figure out the entry point from that.

Reply:
I think npm install and then doing what deployer does (import and then ping operation) would be good.

Reply:
this is for any package (not necessarily a service)?  I was thinking an npm install + importing it from some generated typescript.  then compile & run.  that way you know the types are ok and the entry point is correct.

Reply:
in the new-new world a service is started by `await`ing the default export of the main entry point.  so this would expose an issue with API/service code once that starts rolling out.  also, I feel the primary job of this action is to validate the npm package itself, which is effectively whatever is exported from the root index.ts, vs functionality as a service.  we have other things that do that.

Reply:
that makes total sense

Reply:
this is probably a known limitation using `import`: if the package references excluded(e.g. from *.test.ts) types or values but doesn't export them through its root `index.ts`, then no error will be reported.
on the other hand, using `node` cli to load that package's `index.ts` can expose the problem though.